disorder-jack/src/Disorder/Jack/Property.hs

```diff
@@ -0,0 +1,94 @@
+{-# LANGUAGE NoImplicitPrelude #-}
+{-# LANGUAGE LambdaCase #-}
+module Disorder.Jack.Property (
+    gamble
+  , gambleRender
+  , gambleDisplay
+
+  , shrinking
+
+  , sample
+  , sampleTree
+  , printSample
+
+  -- * QuickCheck re-exports
+  , Property(..)
+  , Testable(..)
+  ) where
+
+import           Data.Foldable (for_, traverse_)
+import           Data.Function (($), (.))
+import           Data.Functor (Functor(..))
+import qualified Data.List as List
+import           Data.String (String)
+import           Data.Text (Text)
+import qualified Data.Text as T
+
+import           Disorder.Jack.Core
+import           Disorder.Jack.Tree
+
+import           System.IO (IO, putStrLn)
+
+import           Text.Show (Show)
+import           Text.Show.Pretty (ppShow)
+
+import qualified Test.QuickCheck as QC
+import           Test.QuickCheck.Gen.Unsafe (promote)
+import           Test.QuickCheck.Property (Testable(..), Property(..), Prop(..), Rose(..))
+import           Test.QuickCheck.Property (joinRose)
+
+
+-- | Ask 'Jack' to generate test cases to exercise the given property.
+gamble :: (Show a, Testable prop) => Jack a -> (a -> prop) -> Property
+gamble jack =
```

Comment:
As discussed should we alias this as `forAll`, seems like jack is a mostly source-compatible drop-in